File: PHP/Film.php
<?php
// Class Film: Blueprint objek data film
class Film {
    // Deklarasi atribut/properti privat milik objek
    private string $idFilm;      // Menyimpan ID unik film
    private string $judul;       // Menyimpan judul film
    private string $genre;       // Menyimpan genre film
    private int $durasiMenit;    // Menyimpan durasi film dalam hitungan menit
    private float $rating;       // Menyimpan rating film (skala 0.0 - 10.0)
    private string $gambar;      // Menyimpan path file gambar poster lokal

    // Constructor: Menginisialisasi nilai awal properti saat objek baru dibuat
    public function __construct(
        string $idFilm = "",
        string $judul = "",
        string $genre = "",
        int $durasiMenit = 0,
        float $rating = 0.0,
        string $gambar = ""
    ) {
        $this->idFilm = $idFilm;
        $this->judul = $judul;
        $this->genre = $genre;
        $this->durasiMenit = $durasiMenit;
        $this->rating = $rating;
        $this->gambar = $gambar;
    }

    // --- Getter & Setter Methods ---

    // Mengambil nilai atribut idFilm
    public function getIdFilm(): string {
        return $this->idFilm;
    }

    // Mengisi atau memperbarui nilai atribut idFilm
    public function setIdFilm(string $idFilm): void {
        $this->idFilm = $idFilm;
    }

    // Mengambil nilai atribut judul
    public function getJudul(): string {
        return $this->judul;
    }

    // Mengisi atau memperbarui nilai atribut judul
    public function setJudul(string $judul): void {
        $this->judul = $judul;
    }

    // Mengambil nilai atribut genre
    public function getGenre(): string {
        return $this->genre;
    }

    // Mengisi atau memperbarui nilai atribut genre
    public function setGenre(string $genre): void {
        $this->genre = $genre;
    }

    // Mengambil nilai atribut durasiMenit
    public function getDurasiMenit(): int {
        return $this->durasiMenit;
    }

    // Mengisi nilai durasiMenit dengan validasi angka positif
    public function setDurasiMenit(int $durasiMenit): void {
        if ($durasiMenit > 0) {
            $this->durasiMenit = $durasiMenit;
        }
    }

    // Mengambil nilai atribut rating
    public function getRating(): float {
        return $this->rating;
    }

    // Mengisi nilai rating dengan validasi rentang 0.0 sampai 10.0
    public function setRating(float $rating): void {
        if ($rating >= 0.0 && $rating <= 10.0) {
            $this->rating = $rating;
        }
    }

    // Mengambil path lokasi file gambar poster
    public function getGambar(): string {
        return $this->gambar;
    }

    // Mengisi atau memperbarui path file gambar poster
    public function setGambar(string $gambar): void {
        $this->gambar = $gambar;
    }
}
?>

File: Python/Film.py
# Class Film digunakan sebagai blueprint data objek film
class Film:
    # Constructor tanpa parameter dan dengan parameter (meng-inisialisasi atribut)
    def __init__(self, idFilm: str = "", judul: str = "", genre: str = "", durasiMenit: int = 0, rating: float = 0.0):
        # Atribut private
        self.__idFilm = idFilm          # ID unik film
        self.__judul = judul            # judul film
        self.__genre = genre            # genre film
        self.__durasiMenit = durasiMenit # durasi film dalam hitungan menit
        self.__rating = rating          # nilai rating film (skala 0.0 - 10.0)

    # Method getter untuk mengembalikan nilai idFilm
    def getIdFilm(self) -> str:
        return self.__idFilm

    # Method setter untuk mengisi atau memperbarui nilai idFilm
    def setIdFilm(self, idFilm: str):
        self.__idFilm = idFilm

    # Method getter untuk mengembalikan nilai judul
    def getJudul(self) -> str:
        return self.__judul

    # Method setter untuk mengisi atau memperbarui nilai judul
    def setJudul(self, judul: str):
        self.__judul = judul

    # Method getter untuk mengembalikan nilai genre
    def getGenre(self) -> str:
        return self.__genre

    # Method setter untuk mengisi atau memperbarui nilai genre
    def setGenre(self, genre: str):
        self.__genre = genre

    # Method getter untuk mengembalikan nilai durasiMenit
    def getDurasiMenit(self) -> int:
        return self.__durasiMenit

    # Method setter untuk mengisi atau memperbarui nilai durasiMenit
    def setDurasiMenit(self, durasiMenit: int):
        # Memeriksa apakah durasi bernilai positif
        if durasiMenit > 0:
            self.__durasiMenit = durasiMenit  # Mengisi atau memperbarui jika valid

    # Method getter untuk mengembalikan nilai rating
    def getRating(self) -> float:
        return self.__rating

    # Method setter untuk mengisi atau memperbarui nilai rating
    def setRating(self, rating: float):
        # Memeriksa apakah rating berada pada rentang 0.0 hingga 10.0
        if 0.0 <= rating <= 10.0:
            self.__rating = rating  # Mengisi atau memperbarui jika valid